python/tvm/auto_scheduler/measure.py

```diff
@@ -624,12 +652,10 @@ def local_build_worker(args):
         The build result of this Builder thread.
     """
     inp, build_func, timeout, verbose = args
-    if build_func == "default":
-        build_func = tar.tar
-    elif build_func == "ndk":
-        build_func = ndk.create_shared
-    else:
-        raise ValueError("Invalid build_func" + build_func)
+    assert build_func == BuildFunc.name, (
+        "BuildFunc.name: " + BuildFunc.name + ", but args is: " + build_func
+    )
+    build_func = BuildFunc.build_func
```

Comment:
@FrozenGene @leowang1225 
I remember that we have used the global variable at first. But it was modified to use serialized args later because of some special reason(which I have forgetten)
So this way may cause some other problem. I'm thinking that just pass the registered function name here as a string and this can be serialized easily.

Also cc @merrymercy @comaniac if you can figure out the reason of not using global variable here.

Reply:
I query this piece of code git history , it is related with this pr: https://github.com/apache/tvm/pull/6671. seems it is about spawn problem

Reply:
@jcf94 
I try use @tvm._ffi.register_func, but the custom build func is a python callable, it is not just function. we need the whole python callable context,

Reply:
Ok, then I dont have other better opinions, I'm going to do some test on arm cpus these days, maybe I'll encounter similar problem like you.

Is it possible to:
```python
use_ndk = True

@tvm._ffi.register_func("special_wrapper")
def _wrapper():
    return cc.cross_compiler(ndk.create_shared, options=ndk_options)

build_func = "special_wrapper"

tuner = auto_scheduler.TaskScheduler(tasks, task_weights)
tune_option = auto_scheduler.TuningOptions(
    num_measure_trials=200, # change this to 20000 to achieve the best performance
    builder=auto_scheduler.LocalBuilder(build_func=build_func if use_ndk else "default"),
    runner=auto_scheduler.RPCRunner(
        device_key, host=tracker_ip, port=tracker_port, repeat=3, timeout=50
    ),
    measure_callbacks=[auto_scheduler.RecordToFile(log_file)],
)
```

Reply:
yes

Reply:
`
def cross_compiler(
    compile_func, options=None, output_format=None, get_target_triple=None, add_files=None
):
...

    def _fcompile(outputs, objects, options=None):
        all_options = base_options
        if options is not None:
            all_options += options
        compile_func(outputs, objects + add_files, options=all_options, **kwargs)

    if not output_format and hasattr(compile_func, "output_format"):
        output_format = compile_func.output_format
    output_format = output_format if output_format else "so"

    if not get_target_triple and hasattr(compile_func, "get_target_triple"):
        get_target_triple = compile_func.get_target_triple

    _fcompile.output_format = output_format
    _fcompile.get_target_triple = get_target_triple
    return _fcompile
`

Reply:
i think just pass the custom function name doesn't have much benefit. We could just pass
```python
use_ndk = True
build_func = cc.cross_compiler(ndk.create_shared, options=ndk_options)
tuner = auto_scheduler.TaskScheduler(tasks, task_weights)
tune_option = auto_scheduler.TuningOptions(
    num_measure_trials=200, # change this to 20000 to achieve the best performance
    builder=auto_scheduler.LocalBuilder(build_func=build_func if use_ndk else "default"),
    runner=auto_scheduler.RPCRunner(
        device_key, host=tracker_ip, port=tracker_port, repeat=3, timeout=50
    ),
    measure_callbacks=[auto_scheduler.RecordToFile(log_file)],
)    
```